internal/configuration/settings/serverselection.go

```diff
@@ -54,21 +54,33 @@ type ServerSelection struct { //nolint:maligned
 	// be filtered. This is used with VPN Secure.
 	// TODO extend to providers using FreeOnly.
 	PremiumOnly *bool `json:"premium_only"`
+	// OpenVPN contains settings to select OpenVPN servers
+	// and the final connection.
+	OpenVPN OpenVPNSelection `json:"openvpn"`
+	// Wireguard contains settings to select Wireguard servers
+	// and the final connection.
+	Wireguard WireguardSelection `json:"wireguard"`
 	// StreamOnly is true if VPN servers not for streaming should
-	// be filtered. This is used with VPNUnlimited.
+	// be filtered. This is used with ProtonVPN and VPNUnlimited.
 	StreamOnly *bool `json:"stream_only"`
 	// MultiHopOnly is true if VPN servers that are not multihop
 	// should be filtered. This is used with Surfshark.
 	MultiHopOnly *bool `json:"multi_hop_only"`
 	// PortForwardOnly is true if VPN servers that don't support
 	// port forwarding should be filtered. This is used with PIA.
 	PortForwardOnly *bool `json:"port_forward_only"`
-	// OpenVPN contains settings to select OpenVPN servers
-	// and the final connection.
-	OpenVPN OpenVPNSelection `json:"openvpn"`
-	// Wireguard contains settings to select Wireguard servers
-	// and the final connection.
-	Wireguard WireguardSelection `json:"wireguard"`
+	// SecureCoreOnly is true if VPN servers without secure core should
+	// be filtered. This is used with ProtonVPN.
+	SecureCoreOnly *bool `json:"secure_core_only"`
+	// TorOnly is true if VPN servers without tor should
+	// be filtered. This is used with ProtonVPN.
+	TorOnly *bool `json:"tor_only"`
+	// P2POnly is true if VPN servers not for p2p should
+	// be filtered. This is used with ProtonVPN.
+	P2POnly *bool `json:"p2p_only"`
+	// IPv6Only is true if VPN servers without IPv6 support should
+	// be filtered. This is used with ProtonVPN.
+	IPv6Only *bool `json:"ipv6_only"`
```

Comment:
What does this filter do? 🤔 
Because AFAIK there are only IPv4 addresses to reach Protonvpn servers.

Reply:
Not sure! It's indeed not a supported feature by any current server, but I just wanted to implement all features listed in [protonvpn-nm-lib](https://github.com/ProtonVPN/protonvpn-nm-lib/blob/master/protonvpn_nm_lib/enums.py#L44-L49)

Reply:
Removed due to the strange filter option name `IPV6_ONLY` and because it's unused. We can always re-add it later if needed.